fix: renaming a data set to itself minus its rightmost qualifier causes rename to fail (#4295)
zowe/zowe-explorer-vscode: packages/zowe-explorer/CHANGELOG.md

### New features and enhancements

### Bug fixes

> - Fixed an issue where renaming a data set to itself minus its rightmost qualifier caused the rename to fail. [#4273](https://github.com/zowe/zowe-explorer-vscode/issues/4273)

Reviewer: This entry is easy to digest from a developer's perspective, but I'm wondering if we could phrase this in a way that users can immediately understand:

```suggestion
- Fixed an issue that caused a rename to fail when removing the rightmost qualifier from a data set name. [#4273](https://github.com/zowe/zowe-explorer-vscode/issues/4273)
```

Author: Thanks lol, I was struggling to think of good changelog entry there and hoped ana or someone else would help out there 😂
